pkg/backend/metrics.go

```diff
@@ -0,0 +1,203 @@
+package backend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+)
+
+const (
+	metricPackage = "backend.openshiftcluster"
+	metricInstall = metricPackage + ".installphase"
+	enabled       = "Enabled"
+	disabled      = "Disabled"
+	custom        = "Custom"
+	defaultSet    = "Default"
+	unknown       = "unknown"
+	empty         = "empty"
+)
+
+func (ocb *openShiftClusterBackend) emitMetrics(log *logrus.Entry, doc *api.OpenShiftClusterDocument, operationType, provisioningState api.ProvisioningState, backendErr error) {
```

Comment:
I recommend unit testing this function using mock emitters. An example from `pkg/monitor`: https://github.com/Azure/ARO-RP/blob/master/pkg/monitor/cluster/certificateexpirationstatuses_test.go.

Reply:
It is getting unit tested by [openshiftcluster_test.go](https://github.com/Azure/ARO-RP/blob/7caa4a1bc9fb359d966a725a299d0cdff22af8d0/pkg/backend/openshiftcluster_test.go).
Test coverage is about 66%, it could be better, but it's not worth modifying those tests further.
I can add some unit tests specifically for `emitMetrics`.